1_clusters/binary-auth.tf

```diff
@@ -0,0 +1,118 @@
+resource "random_pet" "keyring-name" {
+}
+
+resource "null_resource" "resource-to-wait-on" {
+  provisioner "local-exec" {
+    command = "sleep 60" # one minute to allow eventual consistency of APIs avoiding failures downstream
```

Comment:
Which APIs are eventually consistent?

Reply:
containeranalysis.googleapis.com -- without this waiting, the build fails every time

Reply:
Got it. Could we add a `depends_on` instead or as well?

`depends_on = [module.project-services.project_id]`.

Reply:
Can you help me with what this would help do?  My understanding is that the 'resource-to-wait-on' would wait on the value of project_id, which is a passed-in variable and will return right away.  The 60 seconds is on the Google side for the API to be recognized on the project as being enabled.  The service enablement call isn't blocking (or falsely blocking) and returns back rather quickly, but when immediately used afterwards, the code fails due to the API not being enabled.  Waiting a few seconds (or running a second time) works.

Reply:
The project_id is actually pulled from the project services so it waits until the creation calls return before returning it: https://github.com/terraform-google-modules/terraform-google-project-factory/blob/master/modules/project_services/outputs.tf#L19

If we think some of the services falsely claim they're ready when they're not, I'd say we include both the explicit wait_for (timeout) and the depends_on (which would catch any services which are taking *longer* than 60 seconds to activate).